addons/hr_expense/tests/test_expenses.py

```diff
@@ -437,3 +435,129 @@ def test_form_defaults_from_product(self):
         self.assertEqual(expense.product_uom_id, product.uom_id)
         self.assertEqual(expense.tax_ids, product.supplier_taxes_id)
         self.assertEqual(expense.account_id, product._get_product_accounts()['expense'])
+
+    def test_expense_account(self):
+        """ Checking accounting move entries for the accounts set on the expenses """
+
+        account_expense_1 = self.env['account.account'].create({
+            'code': '610010',
+            'name': 'Expense Account 1'
+        })
+        account_expense_2 = self.env['account.account'].create({
+            'code': '610020',
+            'name': 'Expense Account 2'
+        })
+
+        expense_sheet = self.env['hr.expense.sheet'].create({
+            'name': 'First Expense for employee',
+            'employee_id': self.expense_employee.id,
+            'journal_id': self.company_data['default_journal_purchase'].id,
+            'accounting_date': '2022-01-20',
+            'expense_line_ids': [
+                (0, 0, {
+                    # Expense on Expense Account 1
+                    'name': 'expense_1',
+                    'date': '2022-01-05',
+                    'account_id': account_expense_1.id,
+                    'product_id': self.product_a.id,
+                    'unit_amount': 115.0,
+                    'employee_id': self.expense_employee.id,
+                }),
+                (0, 0, {
+                    # Expense on Expense Account 2
+                    'name': 'expense_2',
+                    'date': '2022-01-08',
+                    'account_id': account_expense_2.id,
+                    'product_id': self.product_a.id,
+                    'unit_amount': 230.0,
+                    'employee_id': self.expense_employee.id,
+                }),
+            ],
+        })
+
+        self.assertRecordValues(expense_sheet, [{'state': 'draft', 'total_amount': 345.0}])
+
+        expense_sheet.action_submit_sheet()
+        expense_sheet.approve_expense_sheets()
+        expense_sheet.action_sheet_move_create()
+
+        # Check expense sheet journal entry values.
+        self.assertRecordValues(expense_sheet.account_move_id.line_ids.sorted('balance'), [
+            # Receivable lines:
+            {
+                'debit': 0.0,
+                'credit': 230.0,
+                'amount_currency': -230.0,
```

Comment:
no need to check for `amount_currency` as you don't have a foreign currency setup. It's just testing that that it's equals to balance.